### PR TITLE
Controller spec

### PR DESCRIPTION
**Pull request by Tommymarvel**
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -1,19 +1,16 @@
 require 'rails_helper'
 
-RSpec.describe 'Posts', type: :request do
-  describe 'GET /index' do
-    it 'returns http success and check correct placeholder text.' do
-      get '/users/:author_id/posts'
-      expect(response).to have_http_status(:ok)
-      expect(response.body).to include('Here is a list of posts for a given user')
-    end
+RSpec.describe 'Posts routing', type: :routing do
```
Comment (OLIPLICHE) [inline spec/requests/posts_spec.rb]: - Please on the Request Spec test, you need to use `type: request` instead.
- Open these links to Learn more: [request-spec](https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec) and [Request Spec](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/articles/request_specs.md)

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-      expect(response).to have_http_status(:ok)
-      expect(response.body).to include('Here show post by ID')
-    end
+  it 'routes to the post show' do
+    expect(get: '/users/1/posts/1').to route_to('posts#show', user_id: '1', id: '1')
+  end
+
+  it 'has named routes for the posts resource' do
+    expect(get: user_posts_path(1)).to be_routable
+    expect(get: user_post_path(1, 1)).to be_routable
```
Comment (OLIPLICHE) [inline spec/requests/posts_spec.rb]: - [ ] Spec file includes tests that ensure the response body includes the correct placeholder text.
Example: expect(response. body).to include("Welcome to user index page")
- Open this links to Learn more: [Request Spec](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/articles/request_specs.md)

File: spec/requests/users_spec.rb [hunk 3]
```diff
@@ -1,19 +1,16 @@
 require 'rails_helper'
 
-RSpec.describe 'Users', type: :request do
-  describe 'GET /index' do
-    it 'returns http success and check correct placeholder text.' do
-      get '/'
-      expect(response).to have_http_status(:ok)
-      expect(response.body).to include('Here is a list of users')
-    end
+RSpec.describe 'Users routing', type: :routing do
```
Comment (OLIPLICHE) [inline spec/requests/users_spec.rb]: - [ ] Please on the Request Spec test, you need to use `type: request` instead.

Open these links to Learn more: [request-spec](https://relishapp.com/rspec/rspec-rails/docs/request-specs/request-spec) and [Request Spec](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/articles/request_specs.md)

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -1,19 +1,16 @@
...
-      expect(response).to have_http_status(:ok)
-      expect(response.body).to include('Here show user by id')
-    end
+  it 'routes to the user show' do
+    expect(get: '/users/1').to route_to('users#show', id: '1')
+  end
+
+  it 'has named routes for the users resource' do
+    expect(get: users_path).to be_routable
+    expect(get: user_path(1)).to be_routable
```
Comment (OLIPLICHE) [inline spec/requests/users_spec.rb]: - [ ] Spec file includes tests that ensure the response body includes the correct placeholder text.

Example: expect(response. body).to include("Welcome to user index page")
- Open this links to Learn more: [Request Spec](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/articles/request_specs.md)

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @Tommymarvel, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Gitflow is correctly used (Pull Request is made from the feature branch into the dev branch) 👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi @Tommymarvel,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
Kindly add the MIT license to your readme :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**